### PR TITLE
#167191395 setup link atom

### PR DESCRIPTION
**Pull request by frankly034**
#### What does this PR do?

- This PR sets up a Link atom with styleguide

#### Description of task to be completed?

- setup link atom
- setup styleguidist for the link
- write tests for the link atom

#### How should this be manually tested?

  - $ git clone https://github.com/andela/king-kong-ah-frontend.git
  - $ npm install
  - $ npm test

#### Any background context you want to provide?

- You would need npm to run this application

#### What are the relevant pivotal tracker stories?

- PT Story link - [#167191395](https://www.pivotaltracker.com/story/show/167191395)

#### Screenshots (if appropriate)

image

### CODE REVIEW

**Review by Sojisoyoye — APPROVED** [review]

**Review by akinyeleolat — APPROVED** [review]

File: src/components/UI/atoms/Link/Link.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+const Link = ({
+  href,
+  isTag,
+  fontSize,
+  margin,
+  onClick,
+  children,
+}) => (
+  <Link.Container
+    href={href || '#'}
```
Comment (benfluleck) [inline src/components/UI/atoms/Link/Link.jsx]: Put this is defaultprops please.

File: src/components/UI/atoms/Link/Link.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    transition: .5s border ease-in-out;
+    border: solid 1px transparent;
+    font-family: 'Roboto';
+    margin: ${margin};
+    font-size: ${theme.fontSizes[fontSize]};
+    color: ${theme.textColors.primary};
+    &:visited {
+      color: ${theme.textColors.primary};
+    }
+    background: ${isTag === true && '#F4EDED'};
```
Comment (benfluleck) [inline src/components/UI/atoms/Link/Link.jsx]: just `isTag && '#F4EDED' `should work

File: src/components/UI/atoms/Link/Link.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    padding: ${isTag === true && '0.8rem'};
+    color: ${isTag === true && theme.textColors.primary};
+    box-shadow: ${isTag === true && '16px 16px 32px rgba(0, 0, 0, 0.05)'};
+    width: ${isTag === true && '25.8rem'};
```
Comment (benfluleck) [inline src/components/UI/atoms/Link/Link.jsx]: Please correct all other instances

File: src/components/UI/atoms/Link/Link.jsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+};
+
+Link.Container = styled.a`
+  ${({
+    theme,
+    isTag,
+    fontSize,
+    margin,
+  }) => `
+    padding: 0;
```
Comment (benfluleck) [inline src/components/UI/atoms/Link/Link.jsx]: I expect logic for internal and external links. e.g. links to other sites vs links to places in your app. Something link this
```
 return isExternal ? (
    <TextLink.External {...linkProps} href={to} target={'_blank'}>
      {innerContainer}
    </TextLink.External>
  ) : (
    <TextLink.Internal {...linkProps} to={to}>
      {innerContainer}
    </TextLink.Internal>
  )
```

File: src/components/UI/atoms/Link/Link.jsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+Link.defaultProps = {
+  display: 'inline-block',
+  fontSize: 'normal',
+  margin: '0',
+};
+
+Link.Container = styled.a`
+  ${({
+    theme,
+    isTag,
```
Comment (benfluleck) [inline src/components/UI/atoms/Link/Link.jsx]: Can a link be disabled? 🤔

File: src/components/UI/atoms/Link/Link.jsx (hunk 1 above)
Comment (benfluleck) [inline src/components/UI/atoms/Link/Link.jsx]: Don't use href use Link from `react-router-dom` so you have access to the apps internal routes. You might need to mock Router for the tests as I don't think you are allowed to render links in isolation

**Review by benfluleck — CHANGES_REQUESTED** [review]

File: src/components/UI/atoms/StyledLink/StyledLink.spec.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import StyledLink from './StyledLink';
+import Title from '../Title/Title';
+import '@testing-library/jest-dom/extend-expect';
+
+afterEach(cleanup);
+
+describe('StyledLink', () => {
+  it('should render the Link component', () => {
+    const { getByText } = render(
+      <MemoryRouter>
```
Comment (benfluleck) [inline src/components/UI/atoms/StyledLink/StyledLink.spec.js]: Please look at your colleagues' tests, I expect you component tests not to have repeated logic

**Review by benfluleck — CHANGES_REQUESTED** [review]
Feedback left

**Review by benfluleck — APPROVED** [review]

**Pull request merged**